src/overlays/actors/ovl_En_Egblock/z_en_egblock.c

```diff
@@ -28,28 +41,185 @@ const ActorInit En_Egblock_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/EnEgblock_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/EnEgblock_Draw.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABAAF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABACB4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABAE64.s")
+void EnEgblock_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    CollisionHeader* colHeader = NULL;
+    EnEgblock* this = THIS;
+
+    DynaPolyActor_Init(&this->dyna, 0);
+    CollisionHeader_GetVirtual(&gEyegoreBlockColHeader, &colHeader);
+    this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
+    this->dyna.actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->inactive = false;
+    Actor_SetScale(&this->dyna.actor, 0.5f);
+    this->type = GET_EGBLOCK_TYPE(&this->dyna.actor);
+    this->param1 = GET_EGBLOCK_PARAM_F80(&this->dyna.actor);
+    this->param2 = GET_EGBLOCK_PARAM_7F(&this->dyna.actor);
+    if (this->type == EGBLOCK_TYPE_ACTIVE) {
+        this->dyna.actor.colChkInfo.health = 1;
+        this->actionFunc = EnEgblock_Active;
+    } else {
+        this->actionFunc = EnEgblock_Static;
+    }
+}
+
+void EnEgblock_Destroy(Actor* thisx, PlayState* play2) {
+    EnEgblock* this = THIS;
+    PlayState* play = play2;
+
+    if (this->dyna.actor.colChkInfo.health == 1) {
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void EnEgblock_Active(EnEgblock* this, PlayState* play) {
+    if (this->dyna.actor.colChkInfo.health <= 0) {
+        s32 i;
+
+        for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+            EnEgblock_SpawnEffect(this, &this->dyna.actor.world.pos, 30, (i & 1) + EGBLOCK_EFFECT_DEBRIS_FLAT);
+            //! @bug Egblock doesn't have an effect with index 2, so half of these spawns do nothing.
+        }
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+        Actor_SpawnFloorDustRing(play, &this->dyna.actor, &this->dyna.actor.world.pos, 30.0f, 30, 10.0f, 100, 30, true);
+        this->inactive = true;
+        this->timer = 50;
+        this->actionFunc = EnEgblock_Inactive;
+    }
+}
+
+void EnEgblock_Inactive(EnEgblock* this, PlayState* play) {
+    if (this->timer == 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    }
+}
+
+void EnEgblock_Static(EnEgblock* this, PlayState* play) {
+}
+
+void EnEgblock_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnEgblock* this = THIS;
+
+    this->actionFunc(this, play);
+    DECR(this->timer);
+
+    EnEgblock_UpdateEffects(this, play);
+}
+
+void EnEgblock_Draw(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnEgblock* this = THIS;
+
+    func_8012C28C(play->state.gfxCtx);
+
+    if (this->inactive != true) {
+        OPEN_DISPS(play->state.gfxCtx);
```

Comment:
Very minor thing, but we prefer to have a blank line between `OPEN_DISPS`/`CLOSE_DISPS` macros and the following/preceding line